tests/integration/targets/rds_instance_modify/tasks/main.yml

```diff
@@ -306,6 +306,168 @@
           - db_info.instances[0].ca_certificate_identifier == "rds-ca-ecc384-g1"
   # Test modifying CA certificate identifier Complete-------------------------------------------
 
+  # Test modifying cloudwatch log exports -------------------------------------------
+    - name: Enable all cloudwatch log exports - check_mode
+      amazon.aws.rds_instance:
+        state: present
+        db_instance_identifier: "{{ modified_instance_id }}"
+        enable_cloudwatch_logs_exports: ["audit", "error", "general", "slowquery"]
+      register: result
+      check_mode: true
+      vars:
+        ansible_python_interpreter: "{{ botocore_virtualenv_interpreter }}"
+
+    - name: Get current cloudwatch log exports
+      amazon.aws.rds_instance_info:
+        db_instance_identifier: "{{ modified_instance_id }}"
+      register: db_info
+    - name: Assert that cloudwatch log exports has been modified - check_mode
+      ansible.builtin.assert:
+        that:
+          - result is changed
+          - result is not failed
+          - db_info.instances[0].enabled_cloudwatch_logs_exports is not defined
+
+    - name: Enable all cloudwatch log exports
+      amazon.aws.rds_instance:
+        state: present
+        db_instance_identifier: "{{ modified_instance_id }}"
+        enable_cloudwatch_logs_exports: ["audit", "error", "general", "slowquery"]
+      register: result
+      vars:
+        ansible_python_interpreter: "{{ botocore_virtualenv_interpreter }}"
+
+    - name: Get current cloudwatch log exports
+      amazon.aws.rds_instance_info:
+        db_instance_identifier: "{{ modified_instance_id }}"
+      register: db_info
+      # It applies immediately but takes a couple seconds
+      until:
+        - db_info.instances[0].db_instance_status == 'available'
+        - not db_info.instances[0].pending_modified_values
+      retries: 10
```

Comment:
Did you try running the previous task with wait: true instead of using retries in this one?

Reply:
`wait: true` is the default, I didn't try explicitly adding it no.